WordPress/Classes/Services/NotificationSyncMediator.swift

```diff
@@ -181,7 +181,19 @@ final class NotificationSyncMediator: NotificationSyncMediatorProtocol {
     ///     - completion: Callback to be executed on completion.
     ///
     func markAsUnread(_ notification: Notification, completion: ((Error?)-> Void)? = nil) {
-        mark([notification], asRead: false, completion: completion)
+        markAsUnread([notification], completion: completion)
+    }
+
+    /// Marks a Notification as Unread.
+    ///
+    /// - Note: This method should only be used on the main thread.
+    ///
+    /// - Parameters:
+    ///     - notifications: The notifications that should be marked unread.
+    ///     - completion: Callback to be executed on completion.
+    ///
+    func markAsUnread(_ notifications: [Notification], completion: ((Error?)-> Void)? = nil) {
```

Comment:
Can we not force this by prefixing `@MainActor` or Dispatching to main inside of the function? If it **must** be used from the main thread and that's the only use case, we could avoid possible misuses by doing so.

Reply:
@alpavanoglu Why do you think that method **must** be executed on the main thread. Is it because of the Core Data object `Notification`?

Reply:
Because of the following comment:
`- Note: This method should only be used on the main thread.`